### DISCUSSION TITLE
리팩터링 2주차 안건

### DISCUSSION THREAD

**Discussion by gitul0515** [prgrms-web-devcourse/Team-BackFro-ArtZip-FE#242] [02.Refactoring]
# 중요

소셜 로그인에 성공하면 oauth/callback 페이지로 리다이렉트 된 후, 
쿼리 파라미터로 받아온 토큰을 저장하는 방식을 취하고 있습니다. 

![image](https://user-images.githubusercontent.com/80658269/187028811-6d64a0b9-a4fd-41fe-94d5-79d0a0346096.png)

이 때, 아주 잠깐이지만 URL에 토큰이 보이고 있습니다. 

![image](https://user-images.githubusercontent.com/80658269/187029144-a3c4a92d-9a19-4119-96f1-9bb45feb77ec.png)

보안에 문제가 생길 수 있는 요소입니다. 
소셜 로그인 성공 시, 토큰을 받아오는 방식을 변경해야 합니다. 
일반적으로 토큰과 같이 중요한 정보를 쿼리 파라미터로 넘기지 않는 것으로 알고 있습니다.

**Comment by gitul0515** [comment]
nextConfig에서 rewrites를 설정하면 url의 토큰을 숨길 순 있을 겁니다. 
하지만 되도록이면 쿼리가 아닌 다른 방식을 사용했으면 하는군요.

**Comment by gitul0515** [comment]
전시회 상세 조회 시, 비용(fee)이 들어 있지 않은 것 같습니다. 
![image](https://user-images.githubusercontent.com/80658269/187221257-cc3a647e-65a9-4871-8753-14718231205d.png)

**Comment by punchdrunkard** [comment]
현재 모든 페이지에서 ssr 에서 토큰을 사용하지 않는다면, 이전의 로컬 스토리지 방식을 사용할 수 있습니다.

**Reply by punchdrunkard** [reply]
우선은 라이브러리 먼저 사용 후, 최후의 보루로.. 남겨두겠습니다.

**Comment by gitul0515** [comment]
멘토님의 조언대로 styles/pages 폴더의 구조를 
pages 폴더의 구조와 동일하게 만들기 
![image](https://user-images.githubusercontent.com/80658269/188258426-9ddd05c1-4c8c-4c74-b8eb-75f52f757ed4.png)

**Reply by Hyevvy** [reply]
pages 폴더 구조와 동일하게 가려면 현재 pages에 있는 style 관련 코드를 같이 옮기는게 좋아보입니다!!

**Reply by punchdrunkard** [reply]
@yunjjeongjo

**Reply by Hyevvy** [reply]
@yunjjeongjo 스타일 pages 폴더에서 styles/pages 폴더로 이사시키는 거 어떻게 생각하십니까 🙃

**Reply by yunjjeongjo** [reply]
어떤식으로 변한다는건지 잘 이해를 못했습니다 😂
폴더 구조만 바꾸는게 맞는거겠죠?? 멘토님 조언이시라면 쨋든 찬성입니다 ~

**Reply by punchdrunkard** [reply]
> 어떤식으로 변한다는건지 잘 이해를 못했습니다 😂
> 
> 폴더 구조만 바꾸는게 맞는거겠죠?? 멘토님 조언이시라면 쨋든 찬성입니다 ~

간단하게 커뮤니티 페이지를 예로 든다면
Styles/pages/communityPage/styles.css
와 같은 방식으로 만든다는 거에요! 
이해가 되시나용?ㅎㅎ

**Reply by yunjjeongjo** [reply]
앗 넵 설명 감사합니다 !! ㅎㅎ

**Comment by yunjjeongjo** [comment]
> # 중요
> 소셜 로그인에 성공하면 oauth/callback 페이지로 리다이렉트 된 후, 쿼리 파라미터로 받아온 토큰을 저장하는 방식을 취하고 있습니다.
> 이 때, 아주 잠깐이지만 URL에 토큰이 보이고 있습니다.
> 보안에 문제가 생길 수 있는 요소입니다. 소셜 로그인 성공 시, 토큰을 받아오는 방식을 변경해야 합니다. 일반적으로 토큰과 같이 중요한 정보를 쿼리 파라미터로 넘기지 않는 것으로 알고 있습니다.

![image](https://user-images.githubusercontent.com/93373357/188369978-1f3934dd-c7f9-480b-a085-85ea55244ab2.png)
@gitul0515 
빈푸에게 온 답장입니다!
리팩토링을 할 예정이시긴 한데 정확한 시기는 저도 알 수 없을것 같네요 ,,
기홍님이 말씀 해주신 Next.js rewrites는 주로 API key를 가지고 request를 보내는 경우에 사용하는 것이라 저희가 쓸 수 없을 것 같아요 ! (제가 잘 모르는걸지도 ..??)
지금 저희는 따로 request를 보내는게 아니라, 백엔드에서 제공해준 링크로 이동해서 로그인을 하면 요청이 가고 백엔드 쪽에서 쿼리로 토큰을 담아서 리다이렉트 시켜주고 있는거라서 ,,
일단 중요한 정보가 없다고 하시니 다행인것 같네용. 그래도 마냥 기다릴 수는 없으니 방법을 좀 더 생각해 보겠습니당

**Reply by punchdrunkard** [reply]
아무리 탈취를 감안하고 만든다고 해도 URL에 대놓고 `refreshToken=~` 이 나오는 건
아무래도 너무 신경을 안 쓴 느낌이 큽니다........ㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋ
`Next.js rewrites` 제가 써봤는데, 윤정님 말씀이 맞는거 같아요!

신경써주셔서 고마워요 Green💚
